### ISSUE TITLE
Overriding a native method does not display a warning for addons

### ISSUE THREAD

**Issue by Pshy0** [godotengine/godot#94176]
### Tested versions

- Reproducible in v4.3.beta2.official [b75f0485b]

### System information

Ubuntu 24.04 LTS

### Issue description

In plugins, when overriding a native built-in method of a node in a derived class, no warning is displayed to indicate that this is not permitted and that the method will never be called.
![image](https://github.com/godotengine/godot/assets/44928041/d4a8a561-c1b6-42d5-83bb-43f36d80a650)
Warnings are displayed in the terminal if you start godot from it, but not in the editor.

### Steps to reproduce

- Create a plugin.
- Add a custom class that extends any node.
- Override any native method of the base node: No warning is displayed.
- Reload project: Warnings are displayed in the terminal if this is how you started godot, but not in the editor.

### Minimal reproduction project (MRP)

[mrp-overriding-native-method-no-warnings.zip](https://github.com/user-attachments/files/16162826/mrp-overriding-native-method-no-warnings.zip)

**Comment by dalexeev** [comment]
This is intentional. All warnings are not produced for addons if the `exclude_addons` setting is enabled. But we could add a setting for exceptions. See:

* https://github.com/godotengine/godot/issues/92394#issuecomment-2132387905
* #93889

**Comment by Pshy0** [comment]
It would be less confusing if, regardless of those settings and exceptions, warnings were always displayed for open files, or at least the file currently tabbed in the editor.

**Comment by huwpascoe** [comment]
Seems odd to have that sort of thing disabled by default. Is there no way for add-ons to suppress warnings intentionally instead?

**Comment by Calinou** [comment]
> It would be less confusing if, regardless of those settings and exceptions, warnings were always displayed for open files, or at least the file currently tabbed in the editor.

I agree this is how it should work, but it might be nontrivial to implement. The GDScript compiler would need to emit warnings regardless of the file path (this is skipped right now), but *implementations* would have to selectively ignore the warning based on the file path.

**Comment by Maaack** [comment]
I was looking to report this same issue - with a slightly confusing twist.

The `show()` native method override actually works from the `addons/` folder (Edit: and beyond). It appears to have been working since at least October 2025.

I am currently on Godot 4.7.2.stable. I don't see any warnings or errors in my terminal when running or opening the project, either. I would like to know if this functionality is intentional or a bug.

Edit: I changed the debug setting `Native Method Override` to `Warn`, and that was enough to get overrides of `show()` to work even from the project root.

**Comment by Maaack** [comment]
Based on the error message and what I've read in the documentation, overriding `show()` should not be working at all. I can open a new issue for this and provide an MRP, if that is the next best course of action.

**Comment by dalexeev** [comment]
> The `show()` native method override actually works from the `addons/` folder

@Maaack Scripts in the `res://addons/` folder ignore all warnings by default (including those treated as errors). You can include or exclude any directory using the `debug/gdscript/warnings/directory_rules` setting. Included directories use the warning configuration specified in Project Settings. Excluded directories ignore all warnings.

Overriding native non-virtual methods doesn't work properly, regardless of the warning configuration. You are just shadowing the native method with a custom one. This may work as intended in some cases, but the override won't be called internally by the engine (when C++ directly calls the method) and may not even be called by GDScript itself (due to static typing optimizations).

**Comment by Maaack** [comment]
@dalexeev Thanks for the clarification!
